pandas/core/window/ewm.py

```diff
@@ -291,7 +291,7 @@ def __init__(
         adjust: bool = True,
         ignore_na: bool = False,
         axis: Axis = 0,
-        times: str | np.ndarray | FrameOrSeries | None = None,
+        times: str | np.ndarray | NDFrameT | None = None,
```

Comment:
Unless `obj` and `times` should be the same sub-type of NDFrame, NDFrame might be a better choice here.

Reply:
yep. these problems and the others you've highlighted weren't picked up when we used the alias as many contributors may not have realized that the alias was a typevar and the implications.